api/src/main/java/io/jsonwebtoken/lang/Services.java

```diff
@@ -0,0 +1,98 @@
+import java.util.List;
+import java.util.ServiceLoader;
+
+/**
+ * Helper class for loading services from the classpath, using a {@link ServiceLoader}. Decouples loading logic for
+ * better separation of concerns and testability.
+ */
+public final class Services {
+
+    private Services() {}
+
+    /**
+     * Loads and instantiates all service implementation of the given SPI class and returns them as a List.
+     *
+     * @param spi The class of the Service Provider Interface
+     * @param <T> The type of the SPI
+     * @return An unmodifiable list with an instance of all available implementations of the SPI. No guarantee is given
+     * on the order of implementations, if more than one.
+     */
+    public static <T> List<T> loadAll(Class<T> spi) {
+        Assert.notNull(spi, "Parameter 'spi' must not be null.");
+        ServiceLoader<T> serviceLoader = ServiceLoader.load(spi);
+
+        List<T> implementations = new ArrayList<>();
+
+        for (T implementation : serviceLoader) {
+            implementations.add(implementation);
+        }
+
+        // fail if no implementations were found
+        if (implementations.isEmpty()) {
+            throw new UnavailableImplementationException(spi);
+        }
+
+        return Collections.unmodifiableList(implementations);
+    }
+
+    /**
+     * Loads the first available implementation the given SPI class from the classpath. Uses the {@link ServiceLoader}
+     * to find implementations. When multiple implementations are available it will return the first one that it
+     * encounters. There is no guarantee with regard to ordering.
+     *
+     * @param spi The class of the Service Provider Interface
+     * @param <T> The type of the SPI
+     * @return A new instance of the service.
+     * @throws UnavailableImplementationException When no implementation the SPI is available on the classpath.
+     */
+    public static <T> T loadFirst(Class<T> spi) {
+        Assert.notNull(spi, "Parameter 'spi' must not be null.");
+        ServiceLoader<T> serviceLoader = ServiceLoader.load(spi);
+        if (serviceLoader.iterator().hasNext()) {
+            return serviceLoader.iterator().next();
+        } else {
+            throw new UnavailableImplementationException(spi);
+        }
+    }
+
+    /**
+     * A backward compatibility {@link Services} utility to help migrate away from {@link Classes#newInstance(String)}.
+     * TODO: remove before v1.0
+     * @deprecated use {@link Services} directly
+     */
+    @Deprecated
+    public static final class Legacy {
+
+        /**
+         * Wraps {@code Services.loadFirst} and throws a {@link UnknownClassException} instead of a
+         * {@link UnavailableImplementationException} to retain the previous behavior. This method should be used when
+         * to retain the previous behavior of methods that throw an unchecked UnknownClassException.
+         */
+        public static <T> T loadFirst(Class<T> spi) {
```

Comment:
@lhazlewood What do you think of `Services.Legacy` ?

Reply:
My preference is still `LegacyServices` in the `impl` module - I wouldn't want a new inner class that we're going to soon delete to show up in the public API if we can avoid it.

Reply:
OH, sorry I miss read your comment about `.impl.lang` vs `.lang` I'm on it